File: heap-priority-queue/last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        maxHeap =[-s for s in stones]
        heapq.heapify(maxHeap)
        while len(maxHeap) > 1: 
           max1 = heapq.heappop(maxHeap)
           max2 = heapq.heappop(maxHeap)
           diff = -max1+max2
           heapq.heappush(maxHeap, -diff)
        return -maxHeap[0]

        
        

File: heap-priority-queue/last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        // Use max-heap 
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder()); 
        for (int stone: stones){
            pq.add(stone); 
        }
        while (pq.size() != 1){
            int heaviest = pq.poll();
            int secondHeaviest = pq.poll();
            int diff = heaviest - secondHeaviest;
            if (diff >= 0){
                pq.add(diff); 
            }
        }
        return pq.peek(); 
    }
}